tests/deploy/e2e/test_autostarted.py

```diff
@@ -1131,3 +1200,152 @@ async def check_resource_in_state(expected_resources: int = 1, state: str = "dep
+    for children in start_state.children:
+        assert children.is_running()
+
+    snippetcompiler.setup_for_snippet(
+        """
+import minimalwaitingmodule
+
+a = minimalwaitingmodule::Sleep(name="test_sleep", agent="agent1", time_to_sleep=120)
+""",
+        autostd=True,
+    )
+
+    # Now, let's deploy a resource
+    version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    async def are_resources_deploying(deployed_resources: int = 1) -> bool:
+        result = await client.resource_list(environment, deploy_summary=True)
+        assert result.code == 200
+        summary = result.result["metadata"]["deploy_summary"]
+        deployed = summary["by_state"]["deploying"]
+        return deployed == deployed_resources
+
+    # Wait for this resource to be deployed
+    await retry_limited(are_resources_deploying, 5)
+
+    # Executors are reporting to be deploying before deploying the first executor, we need to wait for them to be sure that
+    # something is moving
+    await wait_for_consistent_children(
+        current_pid=current_pid,
+        should_scheduler_be_defined=True,
+        should_fork_server_be_defined=True,
+        executor_to_be_defined=1,
+    )
+
+    # Retrieve the current processes, we should have more processes than `start_children`
+    children_after_deployment = construct_scheduler_children(current_pid)
+    for children in children_after_deployment.children:
+        assert children.is_running()
+
+    # We want to simulate a crash
+    children_after_deployment.scheduler.kill()
+
+    # We are only waiting for the scheduler to die. The executor should be cleaned up
+    await retry_limited(
+        wait_for_terminated_status,
+        timeout=const.EXECUTOR_GRACE_HARD + 2,
+        current_children=children_after_deployment.children,
+        expected_terminated_process=1,
+    )
+
+    async def wait_for_down_status() -> bool:
+        """
+        Wait for the down status of Agent1
+        """
+        result = await client.get_agents(environment)
+        assert result.code == 200
+        actual_data = result.result["data"]
+        if len(actual_data) != 1:
+            return False
+        return actual_data[0]["status"] == "down"
+
+    await retry_limited(
+        wait_for_down_status,
+        timeout=5,
+    )
+
+    result = await client.get_agents(environment)
+    assert result.code == 200
+    actual_data = result.result["data"]
+    assert len(actual_data) == 1
+    expected_data = {
+        "environment": environment,
+        "last_failover": actual_data[0]["last_failover"],
+        "name": "agent1",
+        "paused": False,
+        "process_id": None,
+        "process_name": None,
+        "status": "down",
+        "unpause_on_resume": None,
+    }
+    assert actual_data[0] == expected_data
```

Comment:
@wouterdb  I added locally this part in the test (for testing purposes only):
```
    await asyncio.sleep(10)
    await retry_limited(
        wait_for_terminated_status,
        timeout=const.EXECUTOR_GRACE_HARD + 2,
        current_children=children_after_deployment.children,
        expected_terminated_process=2,
    )
    await retry_limited(
        wait_for_terminated_status,
        timeout=const.EXECUTOR_GRACE_HARD + 2,
        current_children=children_after_deployment.children,
        expected_terminated_process=3,
    )
```
But this fails because the fork server and the executor seem somehow to survive: 
```
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/hugo/Documents/Inmanta/copies/inmanta-core/tests/utils.py(87)retry_limited()
-> raise AssertionError("Bounded wait failed")
(Pdb) up
> /home/hugo/Documents/Inmanta/copies/inmanta-core/tests/deploy/e2e/test_autostarted.py(1354)test_scheduler_killed()
-> await retry_limited(
(Pdb) children_after_deployment.children
[psutil.Process(pid=29456, name='python', status='terminated', started='11:45:53'), psutil.Process(pid=29495, name='inmanta: multiprocessing fork server', status='sleeping', started='11:45:55'), psutil.Process(pid=29497, name='inmanta: executor process 78f726cec1879275a4cc2fa2c103afb5 - disconnected', status='sleeping', started='11:45:55')]
```
The executor seems dead but not cleaned, is this expected?